nss-day-cohort-60/national-parks-django-api
# Users can get, create, change, and delete wildlife

Description:
Added get all, get by id, get by park_id, create, update, delete functions into wildlife_views.py

Steps to test:
Fetch the md-wildlife-by-park branch and switch to it
Start/restart debugger/server
Open Postman and use to make the following requests:
Send a GET request for all wildlife (example: http://localhost:8000/wildlife). Verify that a 200 status code is received and all wildlife are displayed.

Send a GET request for a single wildlife by id (example: http://localhost:8000/wildlife/3) Verify that a 200 status code is received and that only the wildlife type with that id is displayed.

Send a GET request for all existing wildlife by park_id (example: http://localhost:8000/wildlife?park_id=1)
Verify 200 status code and that all wildlife from that particular park are displayed.

Use PUT to make a change to an existing wildlife type by specifying the id (example: http://localhost:8000/wildlife/1): Verify 204 status code and do GET request for wildlife to make sure the change was made.

In the example below, National Park was added to Joshua Tree:

{
       "id": 1,
       "name": "Mountain Lion",
       "information": "In all of Joshua Tree National Park, few mammals measure up to the majestic mountain lion, in terms of speed, fierceness, and aggressiveness. Known to many by multiple names, such as cougars, puma, and panthers, these nocturnal animals are excellent hunters who favor ambushing their prey. Lurking in the shades of shrubs, a panther exercises patience before pouncing on prey. Ordinarily, panthers would not attack humans, as they do not recognize humans as prey, but other factors could change this. Being territorial, mountain lions will perceive humans as threats during encroachment of their territories. In such instances, they may attack. Mountain lion attacks on humans are rare as they are very reclusive and avoid human contact.",
       "wildlife_group": 1,
       "image": "https://res.cloudinary.com/dcuwovsbv/image/upload/v1674244197/Wildlife/Mountain-Lion-Puma-concolor.jpg_wbe7ck.webp"
   }
 
POST a new wildlife type to http://localhost:8000/wildlife.  Example below. Verify 201 status code. Do GET request to verify new post.

{
       "name": "New Animal",
       "information": "Lives in a park",
       "wildlife_group": 1,
       "image": "https://res.cloudinary.com/dcuwovsbv/image/upload/v1674244197/Wildlife/Mountain-Lion-Puma-concolor.jpg_wbe7ck.webp"
 
   }
Send a DELETE request for the id of the POST request for the wildlife you just sent. Verify that 204 status code is received
all wildlife and verify it was deleted.
